src/main/kotlin/io/meshcloud/dockerosb/service/GenericServiceInstanceService.kt

```diff
@@ -56,51 +59,55 @@ class GenericServiceInstanceService(
       context.attemptToRefreshRemoteChanges()
 
       val repository = context.buildServiceInstanceRepository()
-      val instanceStatus = repository.getServiceInstanceStatus(request.serviceInstanceId)
+
+      val instance = repository.tryGetServiceInstance(request.serviceInstanceId)
+          ?: throw ServiceInstanceDoesNotExistException(request.serviceInstanceId)
+
+      val instanceStatus = repository.getServiceInstanceStatus(serviceInstanceId = request.serviceInstanceId)
 
       return Mono.just(
           GetLastServiceOperationResponse.builder()
               .operationState(instanceStatus.toOperationState())
               .description(instanceStatus.description)
+              .deleteOperation(instance.deleted)
```

Comment:
I am not sure, whether this is fine. The docs say 
```Set a boolean value indicating whether the current asynchronous operation is a delete operation. Should be
set to true in response to a request for the status of an asynchronous delete request, and
false otherwise.
```
Because it says that this should be true only for the current ongoing delete request, I am not sure whether this should be set to true always for deleted service instances. On the other hand the delete-request should be the last one on any instance i assume.
Nevertheless maybe recheck the docs on this flag.

Reply:
Our delete requests are always async, so that's why instance.deleted is going to be true even before the status will also be "suceeded". The way I read this is that the response means "the last operation on the instance is a delete operation, and its status is "in progress|succeeded|failed". 

I think this is fine as you have pointed out the delete operation is always the last operation in a service instance lifecycle, so I cannot come up with a "counterexample" where this flag would indicate a potentially wrong state?

Reply:
Our delete requests are always async, so that's why instance.deleted is going to be true even before the status will also be "suceeded". The way I read this is that the response means "the last operation on the instance is a delete operation, and its status is "in progress|succeeded|failed".

I think this is fine as you have pointed out the delete operation is always the last operation in a service instance lifecycle, so I cannot come up with a "counterexample" where this flag would indicate a potentially wrong state?